lib/wasix/src/syscalls/wasix/sock_send.rs

```diff
@@ -25,7 +25,20 @@ pub fn sock_send<M: MemorySize>(
     si_flags: SiFlags,
     ret_data_len: WasmPtr<M::Offset, M>,
 ) -> Result<Errno, WasiError> {
-    sock_send_internal(ctx, sock, si_data, si_data_len, si_flags, ret_data_len)
+    let env = ctx.data();
+    let fd_entry = env.state.fs.get_fd(sock).unwrap();
+    let inode = fd_entry.inode.clone();
+    let guard = inode.read();
+    let use_write = match guard.deref() {
+        Kind::Pipe { .. } => true,
+        _ => false,
+    };
```

Comment:
You can probably use `matches!(guard.deref(), Kind::Pipe { .. })`. It's simpler :)

Reply:
done